cloud/scope/machine.go

```diff
@@ -367,6 +367,24 @@ func (m *MachineScope) InstanceSpec() *compute.Instance {
 			instance.ShieldedInstanceConfig.EnableIntegrityMonitoring = false
 		}
 	}
+	if m.GCPMachine.Spec.OnHostMaintenance != nil {
+		switch *m.GCPMachine.Spec.OnHostMaintenance {
+		case infrav1.HostMaintenancePolicyMigrate:
+			instance.Scheduling.OnHostMaintenance = "MIGRATE"
+		case infrav1.HostMaintenancePolicyTerminate:
+			instance.Scheduling.OnHostMaintenance = "TERMINATE"
+		default:
+			// Log a warning that an unknown type was met/return an error
```

Comment:
This was a suggestion, I don't know what the logging infrastructure is in this package, can you add some logging? Or an error? Perhaps @richardcase can weigh in with a suggestion as I think they know this project better than I do

Reply:
Yea, took me a minute to notice that :smirk: 
Updated

Reply:
Hmm, I passed logger to the method but it seems that the apidiff doesn't like it.
Updating...